### PR TITLE
Feature/expansion remove logic

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**

### CODE REVIEW

File: notary-commons/src/main/kotlin/com/d3/commons/expansion/ExpansionUtils.kt [hunk 1]
```diff
@@ -5,15 +5,46 @@
...
     private val gson = Gson()
 
+    /**
+     * Expansion logic that adds signature to the account passed in [expansionDetails]
+     * @param irohaAPI - iroha API
+     * @param mstAccount - credentials of account that adds signatories
+     * @param expansionDetails - expansions details that contain account id to expand
+     * @param triggerTime - time of transaction that is used in multisig
+     */
+    fun addSignatureExpansionLogic(
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/expansion/ExpansionUtils.kt]: Why it's called 'logic'?

File: notary-commons/src/main/kotlin/com/d3/commons/expansion/ExpansionUtils.kt [hunk 2]
```diff
@@ -5,15 +5,48 @@
...
+                Transaction.builder(expansionDetails.accountIdToExpand)
+                    .addSignatory(
+                        expansionDetails.accountIdToExpand,
+                        String.unHex(expansionDetails.publicKey.toLowerCase())
+                    )
+                    .setAccountQuorum(mstAccount.accountId, expansionDetails.quorum)
+                    .setQuorum(consumer.getConsumerQuorum().get())
+                    .setCreatedTime(triggerTime)
+                    .build()
+            ).failure { throw it }
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/expansion/ExpansionUtils.kt]: I'd rather add some logs if expansion completed successfully

File: notary-commons/src/main/kotlin/com/d3/commons/expansion/ExpansionUtils.kt [hunk 3]
```diff
@@ -5,15 +5,51 @@
...
+                    .addSignatory(
+                        expansionDetails.accountIdToExpand,
+                        String.unHex(expansionDetails.publicKey.toLowerCase())
+                    )
+                    .setAccountQuorum(mstAccount.accountId, expansionDetails.quorum)
+                    .setQuorum(consumer.getConsumerQuorum().get())
+                    .setCreatedTime(triggerTime)
+                    .build()
+            ).fold(
+                { hash -> logger.info { "Expansion transaction with hash $hash was sent" } },
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/expansion/ExpansionUtils.kt]: Please, log the name of the account as well.

File: notary-commons/src/main/kotlin/com/d3/commons/expansion/ServiceExpansion.kt [hunk 4]
```diff
@@ -58,72 +58,10 @@ class ServiceExpansion(
...
-                val account =
-                    accountsToExpand.find { account -> account.accountId == expansionDetails.accountIdToExpand }
-                if (account != null) {
-                    // Expand account
-                    expandAccount(account, expansionDetails, block.blockV1.payload.createdTime)
-                    additionalLogic(expansionDetails)
-                } else {
-                    logger.warn("No account with id ${expansionDetails.accountIdToExpand} was found. Cannot expand.")
-                }
+                expansionLogic(expansionDetails, txHash, block.blockV1.payload.createdTime)
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/expansion/ServiceExpansion.kt]: Why we need `txHash`?

Comment (Alexey-N-Chernyshov) [reply notary-commons/src/main/kotlin/com/d3/commons/expansion/ServiceExpansion.kt]: To check tx in Ethereum expansion.

**Pull request merged**